### PR TITLE
Add iOS style dark/light theme toggle

### PR DESCRIPTION
**Pull request by gc**
This is a cleaned up version of #44 with the requested iOS style toggle.

Adds a toggle which allows users to use a dark theme (persists in local storage), for light theme users there should be virtually no change apart from the addition of the toggle.

Live Demo: https://frosty-kirch-4a3ba9.netlify.com/

### CODE REVIEW

File: src/components/Bio.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import React from 'react'
 import 'typeface-montserrat'
 import 'typeface-merriweather'
 
-import profilePic from './profile-pic.jpg'
+import profilePic from '../assets/profile-pic.png'
```
Comment (gaearon) [inline src/components/Bio.js]: Why change to png?

File: src/components/Layout.js [hunk 2]
```diff
@@ -52,7 +57,17 @@ class Layout extends React.Component {
...
     }
+  }
+  render() {
+    const { children } = this.props
     return (
+      <ThemeContext.Consumer>
+      {({theme, toggleTheme }) => (
+      <div style={{
+        color: theme.primary.text.normal,
+        background: theme.primary.background,
```
Comment (gaearon) [inline src/components/Layout.js]: Let's add `transition: 'color 0.2s ease-out, background 0.2s ease-out'`

File: src/components/Style.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        .react-toggle--checked .react-toggle-thumb {
+          border-color: #4f5b62;
+        }
+
+        .react-toggle--checked .react-toggle-track {
+          background-color: #4f5b62;
+        }
+
+        .react-toggle-track {
+          background-color: #b3e5fc;
```
Comment (gaearon) [inline src/components/Style.js]: I'd like to make this black.

File: src/components/Layout.js [hunk 4]
```diff
@@ -61,9 +76,19 @@ class Layout extends React.Component {
           padding: `${rhythm(1.5)} ${rhythm(3 / 4)}`,
         }}
       >
-        {header}
+        <div style={{display: 'flex', justifyContent: 'space-between', alignItems: 'baseline'}}>
+        {this.renderHeader(theme)}
+        <Toggle
+          icons={{checked: <img src={moon}/>, unchecked: <img src={sun}/>}}
```
Comment (gaearon) [inline src/components/Layout.js]: Let's add alts to imgs.

File: src/components/ThemeContext.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    id: 'dark',
+    header: '#ffffff',
+    inlineCode: {
+      background: 'rgba(63,63,63,1)',
+      color: '#e6e6e6',
+    },
+    primary: {
+      background: '#000000',
+      text: {
+        normal: 'rgba(255, 255, 255, 0.8)',
```
Comment (gaearon) [inline src/components/ThemeContext.js]: 0.9

**Comment by gc** [comment]
requested changes have been made

https://frosty-kirch-4a3ba9.netlify.com/

**Comment by j-f1** [comment]
Swiping the switch on iOS doesn’t work properly :/

**Comment by gaearon** [comment]
Let's fix iOS (maybe just fork the toggle and customize it in our own code?)

Also first render animates if we "remember" the dark mode from storage. Can first render be sync with correct mode so we don't transition?

**Comment by gaearon** [comment]
Also there's still blue hover — maybe let's make it something like `#222`?

**Comment by gaearon** [comment]
`react-toggle` seems pretty buggy :-(. I looked into the code but there's a lot of mutation going on and I'm not confident I can understand that logic and fix it. https://github.com/aaronshaf/react-toggle/issues/133

**Comment by j-f1** [comment]
Have you considered using Netlify @gaearon? It would be perfect for PRs like this since it automatically builds a preview for each PR.

**Comment by gaearon** [comment]
Maybe. We'll see.

**Pull request merged**